tools/sysctlfix/main.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	"github.com/spf13/pflag"
+)
+
+// This tool attempts to write a sysctl config file to the sysctl config directory with the highest precedence so
+// we can overwrite any other config and ensure correct sysctl options for Cilium to function.
+
+var (
+	flagSet = pflag.NewFlagSet(os.Args[0], pflag.ContinueOnError)
+
+	sysctlD = flagSet.String("sysctl-conf-dir", "/etc/sysctl.d/", "Path to the sysctl config directory")
+	// The 99-zzz prefix ensures our config file gets precedence over most if not all other files.
+	ciliumOverwrites = flagSet.String(
+		"sysctl-config-file",
+		"99-zzz-override_cilium.conf",
+		"Filename of the cilium sysctl overwrites config file",
+	)
+	// Name of the systemd-sysctl unit to restart after making changes
+	sysctlUnit = flagSet.String(
+		"systemd-sysctl-unit",
+		"systemd-sysctl.service",
+		"Name of the systemd sysctl unit to reload",
+	)
+)
+
+var sysctlConfig = strings.Join([]string{
+	"# Disable rp_filter on Cilium interfaces since it may cause mangled packets to be dropped",
+	"net.ipv4.conf.lxc*.rp_filter = 0",
+	"net.ipv4.conf.cilium_*.rp_filter = 0",
+	"",
+}, "\n")
+
+// This program is executed by an init container so we purposely don't
+// exit with any error codes. In case of errors, the function will print warnings,
+// but we don't block cilium agent pod from running.
+func main() {
+	err := flagSet.Parse(os.Args[1:])
+	if err != nil {
+		fmt.Printf("parse flags: %s\n", err)
+		return
+	}
+
+	info, err := os.Stat(*sysctlD)
+	if err != nil {
+		fmt.Printf("can't stat sysctl.d dir '%s': %s\n", *sysctlD, err)
+		return
+	}
+
+	if !info.IsDir() {
+		fmt.Printf("'%s' is not a directory\n", *sysctlD)
+		return
+	}
+
+	overwritesPath := path.Join(*sysctlD, *ciliumOverwrites)
+	f, err := os.OpenFile(overwritesPath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		fmt.Printf("unable to create cilium sysctl overwrites config: %s\n", err)
+		return
+	}
+	defer f.Close()
+
+	currentContents, err := io.ReadAll(f)
+	if err != nil {
+		fmt.Printf("read config: %s\n", err)
+		return
+	}
+
+	if string(currentContents) == sysctlConfig {
+		fmt.Println("sysctl config up-to-date, nothing to do")
+		return
+	}
+
+	// Move offset into the file to the beginning
+	_, err = f.Seek(0, 0)
```

Comment:
Let's use a named constant for `whence` which should also make the comment superfluous:

```suggestion
	_, err = f.Seek(0, io.SeekStart)
```

Reply:
That is neat, didn't know about those constants